src/io/ioctl.rs

```diff
@@ -75,3 +78,23 @@ pub fn ioctl_fionread<Fd: AsFd>(fd: Fd) -> io::Result<u64> {
         ioctl::ioctl(fd, ctl).map(|n| n as u64)
     }
 }
+
+/// `ioctl(fd, FS_IOC_GETFLAGS)`—Returns the [inode flags] attributes
+///
+/// [inode flags]: https://man7.org/linux/man-pages/man2/ioctl_iflags.2.html
+#[cfg(any(target_os = "android", target_os = "linux"))]
+#[inline]
+#[doc(alias = "FS_IOC_GETFLAGS")]
+pub fn ioctl_getflags<Fd: AsFd>(fd: Fd) -> io::Result<IFlags> {
+    backend::io::syscalls::ioctl_get_flags(fd.as_fd())
+}
+
+/// `ioctl(fd, FS_IOC_SETFLAGS)`—Modify the [inode flags] attributes
+///
+/// [inode flags]: https://man7.org/linux/man-pages/man2/ioctl_iflags.2.html
+#[cfg(any(target_os = "android", target_os = "linux"))]
+#[inline]
+#[doc(alias = "FS_IOC_GETFLAGS")]
+pub fn ioctl_setflags<Fd: AsFd>(fd: Fd, flags: IFlags) -> io::Result<()> {
+    backend::io::syscalls::ioctl_set_flags(fd.as_fd(), flags)
+}
```

Comment:
Also, since the time when the PR was updated, rustix has switched to a new `ioctl::Ioctl` mechanism; see the `ioctl::Ioctl` impls in src/io/ioctl.rs and src/fs/ioctl.rs for examples. Would you be able to adapt the implementation here to use that new mechanism?

Reply:
Thanks for pointing this out, it makes the implementation much more concise.

I've tried to limit duplication of the `ctl` instantiation but the flags have different types in each backend (`u_long` in `libc`, `u32` in `linux_raw_sys`).